warehouse/warehouse.go

```diff
@@ -1678,7 +1691,7 @@ func startWebHandler(ctx context.Context) error {
 			backendconfig.DefaultBackendConfig.WaitForConfig(ctx)
 			mux.HandleFunc("/v1/process", processHandler)
 			// triggers uploads only when there are pending events and triggerUpload is sent for a sourceId
-			mux.HandleFunc("/v1/warehouse/pending-events", pendingEventsHandler)
+			mux.HandleFunc("/v1/warehouse/pending-events", pendingEventsHandler).Methods("GET")
```

Comment:
We don't have a Methods function for ServerMux. We would need to convert it to Router.
```
mux := mux.NewRouter()
mux.HandleFunc("/v1/warehouse/pending-events", pendingEventsHandler).Methods("POST")
```